osclib/stagingapi.py

```diff
@@ -821,12 +821,15 @@ def rm_from_prj(self, project, package=None, request_id=None,
 
         self._remove_package_from_prj_pseudometa(project, package)
         subprj = self.map_ring_package_to_subject(project, package)
-        delete_package(self.apiurl, subprj, package, force=True, msg=msg)
 
-        for sub_prj, sub_pkg in self.get_sub_packages(package):
+        for sub_prj, sub_pkg in self.get_sub_packages(package, project):
             sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
-            if sub_prj != subprj:  # if different to the main package's prj
-                delete_package(self.apiurl, sub_prj, sub_pkg, force=True, msg=msg)
+            # Skip inner-project links for letter staging
+            if not self.is_adi_project(project) and sub_prj == subprj: continue
```

Comment:
As I changed in #948 it would seem better to use `sub_prj == project` especially given that is the form use for the same check in other methods. Given no difference might as well be consistent.

Reply:
hmm.. I think use `sub_prj == subprj` is correct here, `subprj` had through 'map_ring_package_to_subject` but `project` has not. Think about a scenario: FOO package in ring2 and FOO-bar in ring1.

Reply:
The other locations with this same loop and logic have it the other way. Otherwise, those should likely be changed, but not reason for them to be different.

Reply:
alright, I changed it to https://github.com/openSUSE/osc-plugin-factory/pull/1110/commits/2b5e6a99da9b5a259033737785680d82a49dc4f6 , the same logic in the other way, also used the same variable name.